### ISSUE TITLE
[4.0] Improvement Idea - Upgrade from Fontawesome 4.7 to 5?

### ISSUE THREAD

**Issue by jwiesel** [joomla/joomla-cms#20099]
### Steps to reproduce the issue / Background info
Recently Fontawesome has been upgraded from 4.7 to version 5. Unfortunately the CSS classes have changed as part of this process (e.g. fa to far, fas, fab): https://fontawesome.com/how-to-use/upgrading-from-4

### Expected result / Improvement idea
To avoid being stuck with the old Fontawesome version or introducing compatibility breaks after the release of Joomla 4, it would be great to upgrade to Fontawesome 5 already as shown in the upgrade guide: https://fontawesome.com/how-to-use/upgrading-from-4

### Actual result / As is situation
If I interpret the code-base correctly right now Joomla 4 is building up on Fontawesome 4.7: https://github.com/joomla/joomla-cms/blob/4.0-dev/templates/cassiopeia/css/template.min.css

### System information (as much as possible)
Joomla 4 development branch as of 06.04.2018

### Additional comments
If there are points against upgrading to fontawesome 5 that is fine. I guess for template developers for Joomla 4.x it would be good to know the direction we are going: See also: https://github.com/sniggle/joomla-bootstrap4-template

**Comment by dgrammatiko** [comment]
```php
			// Add JavaScript Frameworks - Placed at the end of the document so the pages load faster
			JHtml::_('jquery.framework');
			$doc->addScript($this->baseurl . '/templates/' . $this->template . '/js/popper.min.js');
			$doc->addScript($this->baseurl . '/templates/' . $this->template . '/js/bootstrap.min.js');
			$doc->addScript($this->baseurl . '/templates/' . $this->template . '/js/template.js');
```
This is definitely **not** the way to do it!

About Font Awesome there is an issue in the template repo already...

**Comment by ReLater** [comment]
> Improvement Idea - Upgrade from Fontawesome 4.7 to 5?

@jwiesel 
Could you please explain what the improvement is / the improvements are?

**Comment by Chaosxmk** [comment]
@ReLater From their blog, it looks like they've rewritten everything to weigh less and still provide additional styles/icons for use. Only downside I'm seeing is the time it would take to upgrade everything to v5.

**Comment by jwiesel** [comment]
@dgrammatiko : Thanks for the remark.
@ReLater : To me the new version has different improvements:
* A lot of new icons: FA 4.7 offered 675 icons, 5.0.10 now offers 989 (free version)
* The icons have been redesigned for better readability (also 16px icons), the CSS code has been rewritten
* More consistency in icon-names: e.g. address-book is available as "solid" or "regular" icon instead of the prefixes & suffixes they had before.

Next to that I think it has another aspect: It's a major FontAwesome release. Staying up-to-date before Joomla 4 is officially out might offer the opportunity to stay part of the update cycle for minor releases in the future

**Comment by ReLater** [comment]
@jwiesel 
Thank you for clarification

**Comment by MaqSaid** [comment]
How would you upgrade Fontawesome 4.7 to 5 for a .net core 2.1  Angular 4 application using webpack? Any other quick tool let me know? My concern is also changing the  fa prefix (has been deprecated in version 5.) to new default **fas** solid style and the **fab style for brands**

**Comment by brianteeman** [comment]
@MaqSaid you are asking on the Joomla cms repository - you should be asking fontawesome directly

**Comment by brianteeman** [comment]
Can be closed see #24648
